Google cloud node group fix

anchapin opened the pull request:
This adds a better method for creating GKE cluster that mimics the approach used for AWS EKS clusters.

anchapin committed:
feat: update README and scripts for GKE cluster setup; adjust storage class parameters in values.yaml

anchapin committed:
fix: update GKE provisioner and parameters in storageclass.yaml; increase NFS server storage size to 110Gi

anchapin committed:
fix: update storage class parameters for Google provider; change to pd-ssd and adjust NFS PVC storage class

luisclv commented:
I successfully tested option 2 to deploy `google/create-gke-cluster-large.sh`

The change successfully creates the two node pools and respective nodes in each one of them.

However, when I try to install the chart (just using the provided values.yaml and executing `helm install openstudio-server-large ./openstudio-server --set provider.name=google`, the workloads are sent but are not schedulable. In the Google Cloud console, I am seeing that `db` is stuck at pending, showing 0/1 replicas available, and showing the following events.

```
Message | Reason | First Seen | Last Seen | Count |  
-- | -- | -- | -- | -- | --
Successfully provisioned volume pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6 | ProvisioningSucceeded | Apr 28, 2025, 3:35:43 PM | Apr 28, 2025, 3:35:43 PM | 1 |  
Waiting for a volume to be created either by the external provisioner 'pd.csi.storage.gke.io' or manually by the system administrator. If volume creation is delayed, please verify that the provisioner is running and correctly registered. | ExternalProvisioning | Apr 28, 2025, 3:35:39 PM | Apr 28, 2025, 3:35:39 PM | 2 |  
External provisioner is provisioning volume for claim "default/db" | Provisioning | Apr 28, 2025, 3:35:39 PM | Apr 28, 2025, 3:35:39 PM | 1 |  
Scaled up replica set db-5d4f5d8848 from 0 to 1 | ScalingReplicaSet | Apr 28, 2025, 3:35:38 PM | Apr 28, 2025, 3:35:38 PM | 1 |  
waiting for first consumer to be created before binding | WaitForFirstConsumer | Apr 28, 2025, 3:35:36 PM | Apr 28, 2025, 3:35:36 PM | 1
```


In kubectl the workloads are stuck in ContainerCreating or Pending:
```
NAME                                                       READY   STATUS              RESTARTS   AGE
db-5d4f5d8848-7hxzv                                        0/1     ContainerCreating   0          10m
openstudio-server-nfs-server-provisioner-575f65b86-bmxkl   0/1     ContainerCreating   0          10m
redis-759776746-wmnps                                      0/1     ContainerCreating   0          10m
rserve-846dc789df-r7hh9                                    0/1     Pending             0          10m
web-background-74bf47486-8rgcs                             0/1     Pending             0          10m
web-d988dbc5d-tlv44                                        0/1     Pending             0          10m
worker-69bc656dc5-8r29j                                    1/1     Running             0          10m
worker-69bc656dc5-h49h4                                    1/1     Running             0          10m
```
I get the following error for the `web`, `web-background` and `r-serve` pods. I also noticed that the persistent disks are created.

0/2 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling.

```JSON
{
  "insertId": "1lgftaff3v95cb",
  "jsonPayload": {
    "action": "Scheduling",
    "reportingComponent": "default-scheduler",
    "message": "0/2 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling.",
    "apiVersion": "v1",
    "lastTimestamp": null,
    "eventTime": "2025-04-28T20:35:38.832914Z",
    "type": "Warning",
    "kind": "Event",
    "involvedObject": {
      "uid": "15edb236-b365-405c-8a04-66ead7625506",
      "apiVersion": "v1",
      "resourceVersion": "6862",
      "name": "rserve-846dc789df-r7hh9",
      "namespace": "default",
      "kind": "Pod"
    },
    "reason": "FailedScheduling",
    "reportingInstance": "default-scheduler-gke-ddbb30a9a809481a9ef6-5413-687d-vm",
    "metadata": {
      "resourceVersion": "688",
      "uid": "44d5c371-a9aa-4f65-98a1-897d0d468d29",
      "namespace": "default",
      "managedFields": [
        {
          "manager": "kube-scheduler",
          "fieldsType": "FieldsV1",
          "operation": "Update",
          "fieldsV1": {
            "f:regarding": {},
            "f:reportingInstance": {},
            "f:reportingController": {},
            "f:type": {},
            "f:eventTime": {},
            "f:note": {},
            "f:action": {},
            "f:reason": {}
          },
          "time": "2025-04-28T20:35:38Z",
          "apiVersion": "events.k8s.io/v1"
        }
      ],
      "creationTimestamp": "2025-04-28T20:35:38Z",
      "name": "rserve-846dc789df-r7hh9.183a95d2f43ee5c2"
    },
    "source": {}
  },
  "resource": {
    "type": "k8s_pod",
    "labels": {
      "project_id": "schneider-electric-modeling",
      "location": "us-central1-a",
      "namespace_name": "default",
      "cluster_name": "openstudio-server",
      "pod_name": "rserve-846dc789df-r7hh9"
    }
  },
  "timestamp": "2025-04-28T20:35:41.613395007Z",
  "severity": "WARNING",
  "logName": "projects/schneider-electric-modeling/logs/events",
  "receiveTimestamp": "2025-04-28T20:35:41.613395007Z",
  "errorGroups": [
    {
      "id": "COea_ObHzvfmtQE"
    }
  ]
}

```

luisclv commented:
> I successfully tested option 2 to deploy `google/create-gke-cluster-large.sh`
> 
> The change successfully creates the two node pools and respective nodes in each one of them.
> 
> However, when I try to install the chart (just using the provided values.yaml and executing `helm install openstudio-server-large ./openstudio-server --set provider.name=google`, the workloads are sent but are not schedulable. In the Google Cloud console, I am seeing that `db` is stuck at pending, showing 0/1 replicas available, and showing the following events.
> 
> ```
> Message | Reason | First Seen | Last Seen | Count |  
> -- | -- | -- | -- | -- | --
> Successfully provisioned volume pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6 | ProvisioningSucceeded | Apr 28, 2025, 3:35:43 PM | Apr 28, 2025, 3:35:43 PM | 1 |  
> Waiting for a volume to be created either by the external provisioner 'pd.csi.storage.gke.io' or manually by the system administrator. If volume creation is delayed, please verify that the provisioner is running and correctly registered. | ExternalProvisioning | Apr 28, 2025, 3:35:39 PM | Apr 28, 2025, 3:35:39 PM | 2 |  
> External provisioner is provisioning volume for claim "default/db" | Provisioning | Apr 28, 2025, 3:35:39 PM | Apr 28, 2025, 3:35:39 PM | 1 |  
> Scaled up replica set db-5d4f5d8848 from 0 to 1 | ScalingReplicaSet | Apr 28, 2025, 3:35:38 PM | Apr 28, 2025, 3:35:38 PM | 1 |  
> waiting for first consumer to be created before binding | WaitForFirstConsumer | Apr 28, 2025, 3:35:36 PM | Apr 28, 2025, 3:35:36 PM | 1
> ```
> 
> In kubectl the workloads are stuck in ContainerCreating or Pending:
> 
> ```
> NAME                                                       READY   STATUS              RESTARTS   AGE
> db-5d4f5d8848-7hxzv                                        0/1     ContainerCreating   0          10m
> openstudio-server-nfs-server-provisioner-575f65b86-bmxkl   0/1     ContainerCreating   0          10m
> redis-759776746-wmnps                                      0/1     ContainerCreating   0          10m
> rserve-846dc789df-r7hh9                                    0/1     Pending             0          10m
> web-background-74bf47486-8rgcs                             0/1     Pending             0          10m
> web-d988dbc5d-tlv44                                        0/1     Pending             0          10m
> worker-69bc656dc5-8r29j                                    1/1     Running             0          10m
> worker-69bc656dc5-h49h4                                    1/1     Running             0          10m
> ```
> 
> I get the following error for the `web`, `web-background` and `r-serve` pods. I also noticed that the persistent disks are created.
> 
> 0/2 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling.
> 
> ```json
> {
>   "insertId": "1lgftaff3v95cb",
>   "jsonPayload": {
>     "action": "Scheduling",
>     "reportingComponent": "default-scheduler",
>     "message": "0/2 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling.",
>     "apiVersion": "v1",
>     "lastTimestamp": null,
>     "eventTime": "2025-04-28T20:35:38.832914Z",
>     "type": "Warning",
>     "kind": "Event",
>     "involvedObject": {
>       "uid": "15edb236-b365-405c-8a04-66ead7625506",
>       "apiVersion": "v1",
>       "resourceVersion": "6862",
>       "name": "rserve-846dc789df-r7hh9",
>       "namespace": "default",
>       "kind": "Pod"
>     },
>     "reason": "FailedScheduling",
>     "reportingInstance": "default-scheduler-gke-ddbb30a9a809481a9ef6-5413-687d-vm",
>     "metadata": {
>       "resourceVersion": "688",
>       "uid": "44d5c371-a9aa-4f65-98a1-897d0d468d29",
>       "namespace": "default",
>       "managedFields": [
>         {
>           "manager": "kube-scheduler",
>           "fieldsType": "FieldsV1",
>           "operation": "Update",
>           "fieldsV1": {
>             "f:regarding": {},
>             "f:reportingInstance": {},
>             "f:reportingController": {},
>             "f:type": {},
>             "f:eventTime": {},
>             "f:note": {},
>             "f:action": {},
>             "f:reason": {}
>           },
>           "time": "2025-04-28T20:35:38Z",
>           "apiVersion": "events.k8s.io/v1"
>         }
>       ],
>       "creationTimestamp": "2025-04-28T20:35:38Z",
>       "name": "rserve-846dc789df-r7hh9.183a95d2f43ee5c2"
>     },
>     "source": {}
>   },
>   "resource": {
>     "type": "k8s_pod",
>     "labels": {
>       "project_id": "schneider-electric-modeling",
>       "location": "us-central1-a",
>       "namespace_name": "default",
>       "cluster_name": "openstudio-server",
>       "pod_name": "rserve-846dc789df-r7hh9"
>     }
>   },
>   "timestamp": "2025-04-28T20:35:41.613395007Z",
>   "severity": "WARNING",
>   "logName": "projects/schneider-electric-modeling/logs/events",
>   "receiveTimestamp": "2025-04-28T20:35:41.613395007Z",
>   "errorGroups": [
>     {
>       "id": "COea_ObHzvfmtQE"
>     }
>   ]
> }
> ```

This is the description of the db pod:

```
llara@Luiss-MacBook-Pro openstudio-server-helm % kubectl describe pod db-5d4f5d8848-7hxzv 
Name:                 db-5d4f5d8848-7hxzv
Namespace:            default
Priority:             1000000
Priority Class Name:  high-priority
Service Account:      default
Node:                 gke-openstudio-server-web-node-group-c205e44f-pd4r/10.128.0.31
Start Time:           Mon, 28 Apr 2025 15:35:44 -0500
Labels:               app=db
                      pod-template-hash=5d4f5d8848
                      release=openstudio-server
Annotations:          <none>
Status:               Pending
IP:                   
IPs:                  <none>
Controlled By:        ReplicaSet/db-5d4f5d8848
Containers:
  mongo-db:
    Container ID:   
    Image:          mongo:6.0.7
    Image ID:       
    Port:           27017/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:     1
      memory:  4Gi
    Environment:
      MONGO_INITDB_ROOT_USERNAME:  openstudio
      MONGO_INITDB_ROOT_PASSWORD:  openstudio
    Mounts:
      /data/db from db (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-hhdhf (ro)
Conditions:
  Type                        Status
  PodReadyToStartContainers   False 
  Initialized                 True 
  Ready                       False 
  ContainersReady             False 
  PodScheduled                True 
Volumes:
  db:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  db
    ReadOnly:   false
  kube-api-access-hhdhf:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason                  Age                 From                     Message
  ----     ------                  ----                ----                     -------
  Normal   Scheduled               38m                 default-scheduler        Successfully assigned default/db-5d4f5d8848-7hxzv to gke-openstudio-server-web-node-group-c205e44f-pd4r
  Normal   SuccessfulAttachVolume  38m                 attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6"
  Warning  FailedMount             71s (x26 over 38m)  kubelet                  MountVolume.MountDevice failed for volume "pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6" : rpc error: code = Internal desc = Failed to format and mount device from ("/dev/disk/by-id/google-pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6") to ("/var/lib/kubelet/plugins/kubernetes.io/csi/pd.csi.storage.gke.io/ef7f6dc6e5598623126b60b5ff0000e7070041dc04677c7eac06d289c81116e6/globalmount") with fstype ("ext4") and options (["vers=4"]): mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t ext4 -o vers=4,defaults /dev/disk/by-id/google-pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6 /var/lib/kubelet/plugins/kubernetes.io/csi/pd.csi.storage.gke.io/ef7f6dc6e5598623126b60b5ff0000e7070041dc04677c7eac06d289c81116e6/globalmount
Output: mount: /var/lib/kubelet/plugins/kubernetes.io/csi/pd.csi.storage.gke.io/ef7f6dc6e5598623126b60b5ff0000e7070041dc04677c7eac06d289c81116e6/globalmount: wrong fs type, bad option, bad superblock on /dev/sdc, missing codepage or helper program, or other error.
       dmesg(1) may have more information after failed mount system call.
```

brianlball commented:
`Warning  FailedMount 71s (x26 over 38m)  kubelet MountVolume.MountDevice failed for volume "pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6" : rpc error: code = Internal desc = Failed to format and mount device from ("/dev/disk/by-id/google-pvc-49d05b1a-ad83-4d89-a6b6-d3c7ab0fada6") to ("/var/lib/kubelet/plugins/kubernetes.io/csi/pd.csi.storage.gke.io/ef7f6dc6e5598623126b60b5ff0000e7070041dc04677c7eac06d289c81116e6/globalmount") with fstype ("ext4") and options (["vers=4"]): mount failed: exit status 32`

anchapin committed:
Removes NFS mount options from storage class

Removes the `mountOptions` configuration from the storage class template.
This change ensures that the storage class does not specify NFS version,
allowing for the default configuration and greater compatibility.

luisclv commented:
I was able to install the chart without an issue after changing the NFS mounting type. I am testing this version over the latest release (3.6.1-1) and it seems much more stable.

Thank you @anchapin !
